[v1.30] Update collections count limit docs

g-despot opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. We can triage your pull request to the best possible team for review if you explain why you're making a change (or linking to a pull request) and what changes you've made.

See our [CONTRIBUTING.md](/CONTRIBUTING.md) for information how to contribute.

Thanks again!
-->

### What's being changed:

Update collections count limit docs.

### Type of change:

<!--Please delete options that are not relevant.-->

- [x] **Documentation** updates (non-breaking change to fix/update documentation)

### How Has This Been Tested?

<!-- Please select all options that apply -->

- [ ] **GitHub action** – automated build completed without errors
- [x] **Local build** - the site works as expected when running `yarn start`

databyjp commented on developers/weaviate/starter-guides/managing-collections/collections-scaling-limits.mdx:
```diff
@@ -143,14 +142,14 @@ Use multi-tenancy when you need to support a large number of tenants and priorit
 #### Challenges
 
-- **Access control complexity**: [Fine-grained access control](/developers/weaviate/configuration/authorization) must be implemented to ensure data isolation between tenants.
+- **Access control complexity**: [Fine-grained access control](../../configuration/authorization.md) must be implemented to ensure data isolation between tenants.
```
I wonder if this should highlight "uniform data schema and collection configuration" (or - something shorter and catchier) as a challenge, given that we have "customizability" as an advantage.

databyjp commented:
Other comments from reviewing the "Scaling limits with collections" page:

> When designing a vector database collection definition (data schema) in Weaviate
Should it be `When defining a collection in Weaviate`? It feels not a part of the data schema imo. 

I would like for us to maybe revisit some of the the "advantages"/"challenges", as I am not sure they are completely differentiated. 

In my opinion, they should mirror each other to help with choice, but I'm left a little confused from reading them.

- It lists "data isolation" as an advantage for both. Can we highlight the differences?
- "Simple queries" (under 'one collection per dataset'): I don't know if I would say either requires a filter. I guess we are talking about:
    - `shop_a = client.collections.get('products_shop_a')` vs
    - `shop_a = client.collections.get('products').with_tenant('shop_a')`
- "Query performance" (under 'one collection per dataset'): Would this be any more efficient with multi-tenant collections? I think any searches across multiple tenants would have similar overheads - AFAIK you can not perform any searches/aggregations etc. across multiple tenants.
- Should we add the opposite of `Customizability` to the challenges section for multi-tenancy?
- "Index scalability" (under `Multi-tenancy architecture`) - would this not be similar in either situations? Each "end user / shop" would get their own indexes, whether it is for single tenant or single collection. 

(The scenario described sounds as though it is comparing one collection per end user vs one tenant per end user. But some of these points sound as though we are listing some differences between one monolithic collection vs a multi-tenant collection here. )

Minor point - Phrasing:
> You hit the collections count limit?
"Have you ..."

> using a single collection to multiple datasets
Maybe "dividing a single collection into multiple datasets"

> Managing thousands or even millions of collections becomes nearly impossible.
Should it be in decreasing order here to go from hard -> easy? e.g. "managing millions or even thousands..."

g-despot commented:
Thanks for the feedback. 

- I would leave "data isolation" in both as it's a key point in multi-tenancy but I didn't want to leave it out from collection per dataset. 
- Removed the "Simple queries" point. 
- "Query performance" 
- Added "Uniform collection definition" as the opposite of "Customizability"
- True, it's similar in either situation, but this is the opposite of "Resource overhead." Here, the user only takes care of setting up one index, which works for all tenants and is why it's called "scalable"
